[exporter/splunkhec] Integration tests failing

### Component(s)

exporter/splunkhec

### Describe the issue you're reporting

[Failing CI/CD link](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8743341054/job/23993768430?pr=32529)

Note that this test hasn't been running for some time, as explained in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32207, so this failure may not be the result of a recent change.

Failure output:
```
Running target 'mod-integration-test' in module 'exporter/splunkhecexporter' as part of group 'exporter-1'
make --no-print-directory -C exporter/splunkhecexporter mod-integration-test
running go integration test ./... in /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter
/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/.tools/gotestsum --rerun-fails=1 --packages="./..." -- -race -timeout 360s -parallel 4 -tags=integration,""
go: downloading github.com/testcontainers/testcontainers-go v0.30.0
go: downloading github.com/containerd/containerd v1.7.12
go: downloading dario.cat/mergo v1.0.0
go: downloading github.com/cpuguy83/dockercfg v0.3.1
go: downloading github.com/moby/patternmatcher v0.6.0
go: downloading github.com/moby/term v0.5.0
go: downloading golang.org/x/exp v0.0.0-20230711023510-fffb14384f22
go: downloading github.com/containerd/log v0.1.0
go: downloading github.com/moby/sys/sequential v0.5.0
go: downloading github.com/morikuni/aec v1.0.0
go: downloading github.com/moby/sys/user v0.1.0
∅  internal/integrationtestutils
∅  internal/metadata
✖  . (2m32.997s)

DONE 187 tests, 5 failures in 158.494s

✖  . (1m48.883s)
✖  . (1m50.798s)
✖  . (1m51.054s)
✖  . (1m51.945s)

=== Failed
make[2]: *** [../../Makefile.Common:142: mod-integration-test] Error 1
=== FAIL: . TestSplunkHecExporter/Events_to_Splunk (0.01s)
make[1]: *** [Makefile:165: exporter/splunkhecexporter] Error 2
make: *** [Makefile:122: gointegration-test] Error 2
    logger.go:146: 2024-04-18T19:15:38.289Z	INFO	Test -> Splunk running at:	{"host": "localhost", "uiPort": "32771", "hecPort": "32770", "managementPort": "32769"}
    integration_test.go:248: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter/integration_test.go:248
        	            				/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter/integration_test.go:378
        	Error:      	Received unexpected error:
        	            	HTTP 403 "Forbidden"
        	Test:       	TestSplunkHecExporter/Events_to_Splunk
        	Messages:   	Must not error while sending Logs data

=== FAIL: . TestSplunkHecExporter/Events_to_Splunk_-_Non_default_index (0.01s)
    logger.go:146: 2024-04-18T19:15:38.299Z	INFO	Test -> Splunk running at:	{"host": "localhost", "uiPort": "32771", "hecPort": "32770", "managementPort": "32769"}
    integration_test.go:248: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter/integration_test.go:248
        	            				/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter/integration_test.go:378
        	Error:      	Received unexpected error:
        	            	HTTP 403 "Forbidden"
        	Test:       	TestSplunkHecExporter/Events_to_Splunk_-_Non_default_index
        	Messages:   	Must not error while sending Logs data

=== FAIL: . TestSplunkHecExporter/Events_to_Splunk_-_metrics (0.01s)
    logger.go:146: 2024-04-18T19:15:38.311Z	INFO	Test -> Splunk running at:	{"host": "localhost", "uiPort": "32771", "hecPort": "32770", "managementPort": "32769"}
    integration_test.go:271: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter/integration_test.go:271
        	            				/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter/integration_test.go:380
        	Error:      	Received unexpected error:
        	            	HTTP 403 "Forbidden"
        	Test:       	TestSplunkHecExporter/Events_to_Splunk_-_metrics
        	Messages:   	Must not error while sending Metrics data

=== FAIL: . TestSplunkHecExporter/Events_to_Splunk_-_traces (0.01s)
    logger.go:146: 2024-04-18T19:15:38.323Z	INFO	Test -> Splunk running at:	{"host": "localhost", "uiPort": "32771", "hecPort": "32770", "managementPort": "32769"}
    integration_test.go:287: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter/integration_test.go:287
        	            				/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter/integration_test.go:382
        	Error:      	Received unexpected error:
        	            	HTTP 403 "Forbidden"
        	Test:       	TestSplunkHecExporter/Events_to_Splunk_-_traces
        	Messages:   	Must not error while sending Trace data

=== FAIL: . TestSplunkHecExporter (115.42s)
Setting up...
2024/04/18 19:13:46 github.com/testcontainers/testcontainers-go - Connected to docker: 
  Server Version: 24.0.9
  API Version: 1.43
  Operating System: Ubuntu 22.04.4 LTS
  Total Memory: 15981 MB
  Resolved Docker Host: unix:///var/run/docker.sock
  Resolved Docker Socket Path: /var/run/docker.sock
  Test SessionID: 1aa341b46ef8860350ecffad19b90bc8d88fb2b5a007d6ee626410ba7b555b66
  Test ProcessID: 7d372baf-d8f7-4a9e-adc7-48c805b76679
2024/04/18 19:13:47 🐳 Creating container for image testcontainers/ryuk:0.7.0
2024/04/18 19:13:47 ✅ Container created: e09fe35f718f
2024/04/18 19:13:47 🐳 Starting container: e09fe35f718f
2024/04/18 19:13:48 ✅ Container started: e09fe35f718f
2024/04/18 19:13:48 🚧 Waiting for container id e09fe35f718f image: testcontainers/ryuk:0.7.0. Waiting for: &{Port:8080/tcp timeout:<nil> PollInterval:100ms}
2024/04/18 19:13:48 🔔 Container is ready: e09fe35f718f
2024/04/18 19:14:20 🐳 Creating container for image splunk/splunk:9.1
2024/04/18 19:14:20 ✅ Container created: 95ed1b621637
2024/04/18 19:14:20 🐳 Starting container: 95ed1b621637
2024/04/18 19:14:20 ✅ Container started: 95ed1b621637
2024/04/18 19:14:20 🚧 Waiting for container id 95ed1b621637 image: splunk/splunk:9.1. Waiting for: &{timeout:0xc0002e8818 Log:Ansible playbook complete, will begin streaming splunkd_stderr.log IsRegexp:false Occurrence:1 PollInterval:100ms}
2024/04/18 19:15:38 🔔 Container is ready: 95ed1b621637
{"level":"info","ts":1713467738.199267,"caller":"splunkhecexporter/integration_test.go:145","msg":"Splunk running at:","host":"localhost","uiPort":32771,"hecPort":32770,"managementPort":32769}
Host value updated successfully!
Host value updated successfully!
Host value updated successfully!
Host value updated successfully!
2024/04/18 19:15:38 URL: https://localhost:32769/services/data/indexes
2024/04/18 19:15:38 Create index request status code: 201
2024/04/18 19:15:38 URL: https://localhost:32769/services/data/indexes
2024/04/18 19:15:38 Create index request status code: 201
2024/04/18 19:15:38 URL: https://localhost:32769/services/data/indexes
2024/04/18 19:15:38 Create index request status code: 201
Index created
Tearing down...
Stopping container
2024/04/18 19:15:38 🐳 Terminating container: 95ed1b621637
2024/04/18 19:15:39 🚫 Container terminated: 95ed1b621637
Removed Docker image: splunk/splunk:9.1
Command output:
Untagged: splunk/splunk:9.1
Untagged: splunk/splunk@sha256:c27c1ce363b57491da3b9ede6e72c3e17ff35cb2e2352025c94222018433b750
Deleted: sha256:f16d39d0b7b3fb25d0575f5b5132e8281fc6ab5637e3e6ab9f1d80d6557de18d
Deleted: sha256:ae6f67b991b11e4872b9cbfb1571c283810ce17c5ed12ca770612367d91f6b51
Deleted: sha256:8791d7b3923ff2a319cecd6bb93aa2f8f5a38100aad0d509f181a2a1972d1a8c
Deleted: sha256:392b6ff88507172167aacc2a6880de9ec864f1393f0cb219e03ce81405462e46
Deleted: sha256:0e037c49ed5a2632e0b97fa97f35c814533aa0d03f444acc9837b611d79f3346
Deleted: sha256:8ead4a37a4868cc755178d0aa37171b7bf73ada4606bc4402ba6c6666efc5096
Deleted: sha256:cd89484b4e7c755aee9e162a9969aa6fe455bc2ec6791d26617673520801947d
Deleted: sha256:316036768e1c40f29ef452ad4b5c0e255f62b399b7bba6026709109d890d9e9f
Deleted: sha256:a9ed152c9b3393fb29d6c6cae47fb775c15c3ebfb484ef6c12091584e2f51780
Deleted: sha256:89e957dfc2ff12abd847876fe714bf358d357e5d1a6c91d3c9ac7087ed64d1d3
Deleted: sha256:612622fe34fcd46b5889784f1162a8351ae47e974edf4fddf77d6fd74e29ce24
Deleted: sha256:d4c62ebc480db16bd16300a54498e6afa46811426006ae567a202a1db65a66ea


=== FAIL: . TestSplunkHecExporter/Events_to_Splunk (re-run 1) (0.01s)
    logger.go:146: 2024-04-18T19:17:29.430Z	INFO	Test -> Splunk running at:	{"host": "localhost", "uiPort": "32775", "hecPort": "32774", "managementPort": "32773"}
    integration_test.go:248: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter/integration_test.go:248
        	            				/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter/integration_test.go:378
        	Error:      	Received unexpected error:
        	            	HTTP 403 "Forbidden"
        	Test:       	TestSplunkHecExporter/Events_to_Splunk
        	Messages:   	Must not error while sending Logs data

=== FAIL: . TestSplunkHecExporter (re-run 1) (108.83s)
Setting up...
2024/04/18 19:15:43 github.com/testcontainers/testcontainers-go - Connected to docker: 
  Server Version: 24.0.9
  API Version: 1.43
  Operating System: Ubuntu 22.04.4 LTS
  Total Memory: 15981 MB
  Resolved Docker Host: unix:///var/run/docker.sock
  Resolved Docker Socket Path: /var/run/docker.sock
  Test SessionID: fb812dcb79c20e41c43353a8123f292398410f0bf4180ad41fa8dfe85eb31e7e
  Test ProcessID: 44fd6341-57f7-4225-8fbe-11485a1367d5
2024/04/18 19:15:43 🐳 Creating container for image testcontainers/ryuk:0.7.0
2024/04/18 19:15:43 ✅ Container created: e4db952c9894
2024/04/18 19:15:43 🐳 Starting container: e4db952c9894
2024/04/18 19:15:43 ✅ Container started: e4db952c9894
2024/04/18 19:15:43 🚧 Waiting for container id e4db952c9894 image: testcontainers/ryuk:0.7.0. Waiting for: &{Port:8080/tcp timeout:<nil> PollInterval:100ms}
2024/04/18 19:15:43 🔔 Container is ready: e4db952c9894
2024/04/18 19:16:11 🐳 Creating container for image splunk/splunk:9.1
2024/04/18 19:16:11 ✅ Container created: 078c816a2a05
2024/04/18 19:16:11 🐳 Starting container: 078c816a2a05
2024/04/18 19:16:12 ✅ Container started: 078c816a2a05
2024/04/18 19:16:12 🚧 Waiting for container id 078c816a2a05 image: splunk/splunk:9.1. Waiting for: &{timeout:0xc0002bcba8 Log:Ansible playbook complete, will begin streaming splunkd_stderr.log IsRegexp:false Occurrence:1 PollInterval:100ms}
2024/04/18 19:17:29 🔔 Container is ready: 078c816a2a05
{"level":"info","ts":1713467849.3432121,"caller":"splunkhecexporter/integration_test.go:145","msg":"Splunk running at:","host":"localhost","uiPort":32775,"hecPort":32774,"managementPort":32773}
Host value updated successfully!
Host value updated successfully!
Host value updated successfully!
Host value updated successfully!
2024/04/18 19:17:29 URL: https://localhost:32773/services/data/indexes
2024/04/18 19:17:29 Create index request status code: 201
2024/04/18 19:17:29 URL: https://localhost:32773/services/data/indexes
2024/04/18 19:17:29 Create index request status code: 201
2024/04/18 19:17:29 URL: https://localhost:32773/services/data/indexes
2024/04/18 19:17:29 Create index request status code: 201
Index created
Tearing down...
Stopping container
2024/04/18 19:17:29 🐳 Terminating container: 078c816a2a05
2024/04/18 19:17:30 🚫 Container terminated: 078c816a2a05
Removed Docker image: splunk/splunk:9.1
Command output:
Untagged: splunk/splunk:9.1
Untagged: splunk/splunk@sha256:c27c1ce363b57491da3b9ede6e72c3e17ff35cb2e2352025c94222018433b750
Deleted: sha256:f16d39d0b7b3fb25d0575f5b5132e8281fc6ab5637e3e6ab9f1d80d6557de18d
Deleted: sha256:ae6f67b991b11e4872b9cbfb1571c283810ce17c5ed12ca770612367d91f6b51
Deleted: sha256:8791d7b3923ff2a319cecd6bb93aa2f8f5a38100aad0d509f181a2a1972d1a8c
Deleted: sha256:392b6ff88507172167aacc2a6880de9ec864f1393f0cb219e03ce81405462e46
Deleted: sha256:0e037c49ed5a2632e0b97fa97f35c814533aa0d03f444acc9837b611d79f3346
Deleted: sha256:8ead4a37a4868cc755178d0aa37171b7bf73ada4606bc4402ba6c6666efc5096
Deleted: sha256:cd89484b4e7c755aee9e162a9969aa6fe455bc2ec6791d26617673520801947d
Deleted: sha256:316036768e1c40f29ef452ad4b5c0e255f62b399b7bba6026709109d890d9e9f
Deleted: sha256:a9ed152c9b3393fb29d6c6cae47fb775c15c3ebfb484ef6c12091584e2f51780
Deleted: sha256:89e957dfc2ff12abd847876fe714bf358d357e5d1a6c91d3c9ac7087ed64d1d3
Deleted: sha256:612622fe34fcd46b5889784f1162a8351ae47e974edf4fddf77d6fd74e29ce24
Deleted: sha256:d4c62ebc480db16bd16300a54498e6afa46811426006ae567a202a1db65a66ea


=== FAIL: . TestSplunkHecExporter/Events_to_Splunk_-_Non_default_index (re-run 1) (0.01s)
    logger.go:146: 2024-04-18T19:19:21.314Z	INFO	Test -> Splunk running at:	{"host": "localhost", "uiPort": "32779", "hecPort": "32778", "managementPort": "32777"}
    integration_test.go:248: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter/integration_test.go:248
        	            				/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter/integration_test.go:378
        	Error:      	Received unexpected error:
        	            	HTTP 403 "Forbidden"
        	Test:       	TestSplunkHecExporter/Events_to_Splunk_-_Non_default_index
        	Messages:   	Must not error while sending Logs data

=== FAIL: . TestSplunkHecExporter (re-run 1) (110.75s)
Setting up...
2024/04/18 19:17:33 github.com/testcontainers/testcontainers-go - Connected to docker: 
  Server Version: 24.0.9
  API Version: 1.43
  Operating System: Ubuntu 22.04.4 LTS
  Total Memory: 15981 MB
  Resolved Docker Host: unix:///var/run/docker.sock
  Resolved Docker Socket Path: /var/run/docker.sock
  Test SessionID: 66a55fb79531173f808d5a6b9ab71a454f5508090357b653088542b894743993
  Test ProcessID: 70914a5f-82ff-4638-bb82-3d11b7ffee20
2024/04/18 19:17:33 🐳 Creating container for image testcontainers/ryuk:0.7.0
2024/04/18 19:17:33 ✅ Container created: db94dcc16801
2024/04/18 19:17:33 🐳 Starting container: db94dcc16801
2024/04/18 19:17:33 ✅ Container started: db94dcc16801
2024/04/18 19:17:33 🚧 Waiting for container id db94dcc16801 image: testcontainers/ryuk:0.7.0. Waiting for: &{Port:8080/tcp timeout:<nil> PollInterval:100ms}
2024/04/18 19:17:33 🔔 Container is ready: db94dcc16801
2024/04/18 19:18:03 🐳 Creating container for image splunk/splunk:9.1
2024/04/18 19:18:03 ✅ Container created: 6da988f3baa3
2024/04/18 19:18:03 🐳 Starting container: 6da988f3baa3
2024/04/18 19:18:03 ✅ Container started: 6da988f3baa3
2024/04/18 19:18:03 🚧 Waiting for container id 6da988f3baa3 image: splunk/splunk:9.1. Waiting for: &{timeout:0xc0001f5368 Log:Ansible playbook complete, will begin streaming splunkd_stderr.log IsRegexp:false Occurrence:1 PollInterval:100ms}
2024/04/18 19:19:21 🔔 Container is ready: 6da988f3baa3
{"level":"info","ts":1713467961.2334607,"caller":"splunkhecexporter/integration_test.go:145","msg":"Splunk running at:","host":"localhost","uiPort":32779,"hecPort":32778,"managementPort":32777}
Host value updated successfully!
Host value updated successfully!
Host value updated successfully!
Host value updated successfully!
2024/04/18 19:19:21 URL: https://localhost:32777/services/data/indexes
2024/04/18 19:19:21 Create index request status code: 201
2024/04/18 19:19:21 URL: https://localhost:32777/services/data/indexes
2024/04/18 19:19:21 Create index request status code: 201
2024/04/18 19:19:21 URL: https://localhost:32777/services/data/indexes
2024/04/18 19:19:21 Create index request status code: 201
Index created
Tearing down...
Stopping container
2024/04/18 19:19:21 🐳 Terminating container: 6da988f3baa3
2024/04/18 19:19:22 🚫 Container terminated: 6da988f3baa3
Removed Docker image: splunk/splunk:9.1
Command output:
Untagged: splunk/splunk:9.1
Untagged: splunk/splunk@sha256:c27c1ce363b57491da3b9ede6e72c3e17ff35cb2e2352025c94222018433b750
Deleted: sha256:f16d39d0b7b3fb25d0575f5b5132e8281fc6ab5637e3e6ab9f1d80d6557de18d
Deleted: sha256:ae6f67b991b11e4872b9cbfb1571c283810ce17c5ed12ca770612367d91f6b51
Deleted: sha256:8791d7b3923ff2a319cecd6bb93aa2f8f5a38100aad0d509f181a2a1972d1a8c
Deleted: sha256:392b6ff88507172167aacc2a6880de9ec864f1393f0cb219e03ce81405462e46
Deleted: sha256:0e037c49ed5a2632e0b97fa97f35c814533aa0d03f444acc9837b611d79f3346
Deleted: sha256:8ead4a37a4868cc755178d0aa37171b7bf73ada4606bc4402ba6c6666efc5096
Deleted: sha256:cd89484b4e7c755aee9e162a9969aa6fe455bc2ec6791d26617673520801947d
Deleted: sha256:316036768e1c40f29ef452ad4b5c0e255f62b399b7bba6026709109d890d9e9f
Deleted: sha256:a9ed152c9b3393fb29d6c6cae47fb775c15c3ebfb484ef6c12091584e2f51780
Deleted: sha256:89e957dfc2ff12abd847876fe714bf358d357e5d1a6c91d3c9ac7087ed64d1d3
Deleted: sha256:612622fe34fcd46b5889784f1162a8351ae47e974edf4fddf77d6fd74e29ce24
Deleted: sha256:d4c62ebc480db16bd16300a54498e6afa46811426006ae567a202a1db65a66ea


=== FAIL: . TestSplunkHecExporter/Events_to_Splunk_-_metrics (re-run 1) (0.01s)
    logger.go:146: 2024-04-18T19:21:13.292Z	INFO	Test -> Splunk running at:	{"host": "localhost", "uiPort": "32783", "hecPort": "32782", "managementPort": "32781"}
    integration_test.go:271: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter/integration_test.go:271
        	            				/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter/integration_test.go:380
        	Error:      	Received unexpected error:
        	            	HTTP 403 "Forbidden"
        	Test:       	TestSplunkHecExporter/Events_to_Splunk_-_metrics
        	Messages:   	Must not error while sending Metrics data

=== FAIL: . TestSplunkHecExporter (re-run 1) (111.01s)
Setting up...
2024/04/18 19:19:24 github.com/testcontainers/testcontainers-go - Connected to docker: 
  Server Version: 24.0.9
  API Version: 1.43
  Operating System: Ubuntu 22.04.4 LTS
  Total Memory: 15981 MB
  Resolved Docker Host: unix:///var/run/docker.sock
  Resolved Docker Socket Path: /var/run/docker.sock
  Test SessionID: 5152ddc7b897ec000cb7a3d827d8914a170311f76bab9ccc238a52924bb8e49c
  Test ProcessID: cf067fbe-71ab-4152-8ae2-e5bd37cd918a
2024/04/18 19:19:24 🐳 Creating container for image testcontainers/ryuk:0.7.0
2024/04/18 19:19:24 ✅ Container created: 1c6470fd8d2f
2024/04/18 19:19:24 🐳 Starting container: 1c6470fd8d2f
2024/04/18 19:19:25 ✅ Container started: 1c6470fd8d2f
2024/04/18 19:19:25 🚧 Waiting for container id 1c6470fd8d2f image: testcontainers/ryuk:0.7.0. Waiting for: &{Port:8080/tcp timeout:<nil> PollInterval:100ms}
2024/04/18 19:19:25 🔔 Container is ready: 1c6470fd8d2f
2024/04/18 19:19:55 🐳 Creating container for image splunk/splunk:9.1
2024/04/18 19:19:55 ✅ Container created: 9d9827723acf
2024/04/18 19:19:55 🐳 Starting container: 9d9827723acf
2024/04/18 19:19:55 ✅ Container started: 9d9827723acf
2024/04/18 19:19:55 🚧 Waiting for container id 9d9827723acf image: splunk/splunk:9.1. Waiting for: &{timeout:0xc000265058 Log:Ansible playbook complete, will begin streaming splunkd_stderr.log IsRegexp:false Occurrence:1 PollInterval:100ms}
2024/04/18 19:21:13 🔔 Container is ready: 9d9827723acf
{"level":"info","ts":1713468073.2049854,"caller":"splunkhecexporter/integration_test.go:145","msg":"Splunk running at:","host":"localhost","uiPort":32783,"hecPort":32782,"managementPort":32781}
Host value updated successfully!
Host value updated successfully!
Host value updated successfully!
Host value updated successfully!
2024/04/18 19:21:13 URL: https://localhost:32781/services/data/indexes
2024/04/18 19:21:13 Create index request status code: 201
2024/04/18 19:21:13 URL: https://localhost:32781/services/data/indexes
2024/04/18 19:21:13 Create index request status code: 201
2024/04/18 19:21:13 URL: https://localhost:32781/services/data/indexes
2024/04/18 19:21:13 Create index request status code: 201
Index created
Tearing down...
Stopping container
2024/04/18 19:21:13 🐳 Terminating container: 9d9827723acf
2024/04/18 19:21:14 🚫 Container terminated: 9d9827723acf
Removed Docker image: splunk/splunk:9.1
Command output:
Untagged: splunk/splunk:9.1
Untagged: splunk/splunk@sha256:c27c1ce363b57491da3b9ede6e72c3e17ff35cb2e2352025c94222018433b750
Deleted: sha256:f16d39d0b7b3fb25d0575f5b5132e8281fc6ab5637e3e6ab9f1d80d6557de18d
Deleted: sha256:ae6f67b991b11e4872b9cbfb1571c283810ce17c5ed12ca770612367d91f6b51
Deleted: sha256:8791d7b3923ff2a319cecd6bb93aa2f8f5a38100aad0d509f181a2a1972d1a8c
Deleted: sha256:392b6ff88507172167aacc2a6880de9ec864f1393f0cb219e03ce81405462e46
Deleted: sha256:0e037c49ed5a2632e0b97fa97f35c814533aa0d03f444acc9837b611d79f3346
Deleted: sha256:8ead4a37a4868cc755178d0aa37171b7bf73ada4606bc4402ba6c6666efc5096
Deleted: sha256:cd89484b4e7c755aee9e162a9969aa6fe455bc2ec6791d26617673520801947d
Deleted: sha256:316036768e1c40f29ef452ad4b5c0e255f62b399b7bba6026709109d890d9e9f
Deleted: sha256:a9ed152c9b3393fb29d6c6cae47fb775c15c3ebfb484ef6c12091584e2f51780
Deleted: sha256:89e957dfc2ff12abd847876fe714bf358d357e5d1a6c91d3c9ac7087ed64d1d3
Deleted: sha256:612622fe34fcd46b5889784f1162a8351ae47e974edf4fddf77d6fd74e29ce24
Deleted: sha256:d4c62ebc480db16bd16300a54498e6afa46811426006ae567a202a1db65a66ea


=== FAIL: . TestSplunkHecExporter/Events_to_Splunk_-_traces (re-run 1) (0.01s)
    logger.go:146: 2024-04-18T19:23:05.931Z	INFO	Test -> Splunk running at:	{"host": "localhost", "uiPort": "32787", "hecPort": "32786", "managementPort": "32785"}
    integration_test.go:287: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter/integration_test.go:287
        	            				/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter/integration_test.go:382
        	Error:      	Received unexpected error:
        	            	HTTP 403 "Forbidden"
        	Test:       	TestSplunkHecExporter/Events_to_Splunk_-_traces
        	Messages:   	Must not error while sending Trace data

=== FAIL: . TestSplunkHecExporter (re-run 1) (111.90s)
Setting up...
2024/04/18 19:21:16 github.com/testcontainers/testcontainers-go - Connected to docker: 
  Server Version: 24.0.9
  API Version: 1.43
  Operating System: Ubuntu 22.04.4 LTS
  Total Memory: 15981 MB
  Resolved Docker Host: unix:///var/run/docker.sock
  Resolved Docker Socket Path: /var/run/docker.sock
  Test SessionID: 44af4807f134d02[166](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8743341054/job/23993768430?pr=32529#step:5:167)a589447c8bb86ed95f7a6b96aedef67da99f04f50ed8ae
  Test ProcessID: 61d3985f-95db-49bd-94a7-8b8ed3e239fd
2024/04/18 19:21:16 🐳 Creating container for image testcontainers/ryuk:0.7.0
2024/04/18 19:21:16 ✅ Container created: d623814d8bbf
2024/04/18 19:21:16 🐳 Starting container: d623814d8bbf
2024/04/18 19:21:17 ✅ Container started: d623814d8bbf
2024/04/18 19:21:17 🚧 Waiting for container id d623814d8bbf image: testcontainers/ryuk:0.7.0. Waiting for: &{Port:8080/tcp timeout:<nil> PollInterval:100ms}
2024/04/18 19:21:17 🔔 Container is ready: d623814d8bbf
2024/04/18 19:21:47 🐳 Creating container for image splunk/splunk:9.1
2024/04/18 19:21:47 ✅ Container created: 785c202d75df
2024/04/18 19:21:47 🐳 Starting container: 785c202d75df
2024/04/18 19:21:47 ✅ Container started: 785c202d75df
2024/04/18 19:21:47 🚧 Waiting for container id 785c202d75df image: splunk/splunk:9.1. Waiting for: &{timeout:0xc0003a07c8 Log:Ansible playbook complete, will begin streaming splunkd_stderr.log IsRegexp:false Occurrence:1 PollInterval:100ms}
2024/04/18 19:23:05 🔔 Container is ready: 785c202d75df
{"level":"info","ts":1713468185.8486612,"caller":"splunkhecexporter/integration_test.go:145","msg":"Splunk running at:","host":"localhost","uiPort":32787,"hecPort":32786,"managementPort":32785}
Host value updated successfully!
Host value updated successfully!
Host value updated successfully!
Host value updated successfully!
2024/04/18 19:23:05 URL: https://localhost:32785/services/data/indexes
2024/04/18 19:23:05 Create index request status code: 201
2024/04/18 19:23:05 URL: https://localhost:32785/services/data/indexes
2024/04/18 19:23:05 Create index request status code: 201
2024/04/18 19:23:05 URL: https://localhost:32785/services/data/indexes
2024/04/18 19:23:05 Create index request status code: 201
Index created
Tearing down...
Stopping container
2024/04/18 19:23:05 🐳 Terminating container: 785c202d75df
2024/04/18 19:23:06 🚫 Container terminated: 785c202d75df
Removed Docker image: splunk/splunk:9.1
Command output:
Untagged: splunk/splunk:9.1
Untagged: splunk/splunk@sha256:c27c1ce363b57491da3b9ede6e72c3e17ff35cb2e2352025c94222018433b750
Deleted: sha256:f16d39d0b7b3fb25d0575f5b5132e8281fc6ab5637e3e6ab9f1d80d6557de18d
Deleted: sha256:ae6f67b991b11e4872b9cbfb1571c283810ce17c5ed12ca770612367d91f6b51
Deleted: sha256:8791d7b3923ff2a319cecd6bb93aa2f8f5a38100aad0d509f181a2a1972d1a8c
Deleted: sha256:392b6ff88507172[167](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8743341054/job/23993768430?pr=32529#step:5:168)aacc2a6880de9ec864f1393f0cb219e03ce81405462e46
Deleted: sha256:0e037c49ed5a2632e0b97fa97f35c814533aa0d03f444acc9837b611d79f3346
Deleted: sha256:8ead4a37a4868cc755178d0aa37171b7bf73ada4606bc4402ba6c6666efc5096
Deleted: sha256:cd89484b4e7c755aee9e162a9969aa6fe455bc2ec6791d26617673520801947d
Deleted: sha256:316036768e1c40f29ef452ad4b5c0e255f62b399b7bba6026709109d890d9e9f
Deleted: sha256:a9ed152c9b3393fb29d6c6cae47fb775c15c3ebfb484ef6c12091584e2f51780
Deleted: sha256:89e957dfc2ff12abd847876fe714bf358d357e5d1a6c91d3c9ac7087ed64d1d3
Deleted: sha256:612622fe34fcd46b5889784f1162a8351ae47e974edf4fddf77d6fd74e29ce24
Deleted: sha256:d4c62ebc480db16bd16300a54498e6afa46811426006ae567a202a1db65a66ea


DONE 2 runs, 195 tests, 13 failures in 605.083s
make[1]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib'
```

--- a/exporter/splunkhecexporter/integration_test.go
+++ b/exporter/splunkhecexporter/integration_test.go
@@ -12,6 +12,7 @@ import (
 	"net/http"
 	"net/url"
 	"os/exec"
+	"path/filepath"
 	"strconv"
 	"testing"
 	"time"
@@ -42,12 +43,7 @@ type SplunkContainerConfig struct {
 
 func setup() SplunkContainerConfig {
 	// Perform setup operations here
-	fmt.Println("Setting up...")
 	cfg := startSplunk()
-	integrationtestutils.CreateAnIndexInSplunk(integrationtestutils.GetConfigVariable("EVENT_INDEX"), "event")
-	integrationtestutils.CreateAnIndexInSplunk(integrationtestutils.GetConfigVariable("METRIC_INDEX"), "metric")
-	integrationtestutils.CreateAnIndexInSplunk(integrationtestutils.GetConfigVariable("TRACE_INDEX"), "event")
-	fmt.Println("Index created")
 	return cfg
 }
 
@@ -105,7 +101,14 @@ func startSplunk() SplunkContainerConfig {
 			"SPLUNK_HEC_TOKEN":  integrationtestutils.GetConfigVariable("HEC_TOKEN"),
 			"SPLUNK_PASSWORD":   integrationtestutils.GetConfigVariable("PASSWORD"),
 		},
-		WaitingFor: wait.ForLog("Ansible playbook complete, will begin streaming splunkd_stderr.log").WithStartupTimeout(2 * time.Minute),
+		Files: []testcontainers.ContainerFile{
+			{
+				HostFilePath:      filepath.Join("testdata", "splunk.yaml"),
+				ContainerFilePath: "/tmp/defaults/default.yml",
+				FileMode:          0644,
+			},
+		},
+		WaitingFor: wait.ForHealthCheck().WithStartupTimeout(5 * time.Minute),
 	}
 
 	container, err := testcontainers.GenericContainer(conContext, testcontainers.GenericContainerRequest{

--- a/exporter/splunkhecexporter/testdata/integration_tests_config.yaml
+++ b/exporter/splunkhecexporter/testdata/integration_tests_config.yaml
@@ -1,20 +1,11 @@
-# SPLUNK DATA
-SPLUNK_IMAGE : "splunk/splunk:9.1"
-HOST : "non"
-
-#credentials:
-USER : "admin"
-PASSWORD : "helloworld"
-
-# ports
-UI_PORT: "8000"
-HEC_PORT: "8088"
-MANAGEMENT_PORT: "8089"
-
-# indexes
-EVENT_INDEX : "sck-otel"
-METRIC_INDEX: "sck-metrics"
-TRACE_INDEX: "sck-traces"
-
-# hec tokens
-HEC_TOKEN : "00000000-0000-0000-0000-000000000000"
+HOST: localhost
+USER: admin
+PASSWORD: helloworld
+UI_PORT: "62025"
+HEC_PORT: "62023"
+MANAGEMENT_PORT: "62024"
+EVENT_INDEX: sck-otel
+METRIC_INDEX: sck-metrics
+TRACE_INDEX: sck-traces
+HEC_TOKEN: 00000000-0000-0000-0000-000000000000
+SPLUNK_IMAGE: splunk/splunk:9.1.2

--- a/exporter/splunkhecexporter/testdata/splunk.yaml
+++ b/exporter/splunkhecexporter/testdata/splunk.yaml
@@ -1,0 +1,28 @@
+splunk:
+  hec:
+    enable: True
+    ssl: True
+    port: 8088
+    token: 00000000-0000-0000-0000-000000000000
+  conf:
+    indexes:
+      directory: /opt/splunk/etc/apps/search/local
+      content:
+        sck-otel:
+          coldPath: $SPLUNK_DB/sck-otel/colddb
+          datatype: event
+          homePath: $SPLUNK_DB/sck-otel/db
+          maxTotalDataSizeMB: 512000
+          thawedPath: $SPLUNK_DB/sck-otel/thaweddb
+        sck-metrics:
+          coldPath: $SPLUNK_DB/sck-metrics/colddb
+          datatype: metric
+          homePath: $SPLUNK_DB/sck-metrics/db
+          maxTotalDataSizeMB: 512000
+          thawedPath: $SPLUNK_DB/sck-metrics/thaweddb
+        sck-traces:
+          coldPath: $SPLUNK_DB/sck-traces/colddb
+          datatype: event
+          homePath: $SPLUNK_DB/sck-traces/db
+          maxTotalDataSizeMB: 512000
+          thawedPath: $SPLUNK_DB/sck-traces/thaweddb

